src/java.base/unix/native/libnet/NetworkInterface.c

```diff
@@ -807,6 +809,8 @@ static jobject createNetworkInterface(JNIEnv *env, netif *ifs) {
 
         (*env)->SetObjectArrayElement(env, addrArr, addr_index++, iaObj);
         addrP = addrP->next;
+        (*env)->DeleteLocalRef(env, iaObj);
+        (*env)->DeleteLocalRef(env, ibObj);
```

Comment:
At the cost of duplicating the delete on ibObj, is it worth locating the DeleteLocalRef to directly after the SetObjectArrayElement (so as to make it a little easier to reason about whether the code has followed the set/delete pattern correctly) ?

Reply:
Yes that makes a lot of sense to me. I've done that (and the other similar suggestion) and force-pushed. Thanks!